src/main/java/com/zhanganzhi/chathub/platforms/discord/DiscordAdaptor.java

```diff
@@ -23,19 +29,38 @@ public void start() {
         chatHub.getLogger().info("Discord enabled");
 
         // build jda
-        jda = JDABuilder.createLight(config.getDiscordToken())
-                .enableIntents(
-                        GatewayIntent.MESSAGE_CONTENT
-                )
-                .addEventListeners(new DiscordReceiver(chatHub))
-                .build();
+        JDABuilder builder = JDABuilder.createLight(config.getDiscordToken())
+            .enableIntents(
+                GatewayIntent.MESSAGE_CONTENT
+            )
+            .addEventListeners(new DiscordReceiver(chatHub));
+
+        if (config.isDiscordProxyEnabled()) {
+            chatHub.getLogger().info("Discord proxy enabled");
+            Proxy proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress(config.getDiscordProxyHost(), config.getDiscordProxyPort()));
+
+            OkHttpClient.Builder clientBuilder = IOUtil.newHttpClientBuilder().proxy(proxy);
+
+            WebSocketFactory factory = new WebSocketFactory();
+            factory.getProxySettings()
+                .setHost(config.getDiscordProxyHost())
+                .setPort(config.getDiscordProxyPort());
+
+            builder.setHttpClientBuilder(clientBuilder);
+            builder.setWebsocketFactory(factory);
+        }
+
+        jda = builder.build();
 
         // commands
         jda.updateCommands().addCommands(
-                Commands
-                        .slash("list", "List online players")
-                        .setGuildOnly(true)
-        ).queue();
+            Commands
+                .slash("list", "List online players")
+                .setGuildOnly(true)
+        ).queue(
+            success -> chatHub.getLogger().info("Slash command registered successfully!"),
+            failure -> chatHub.getLogger().error("Slash command failed, {}", failure.getMessage())
+        );
```

Comment:
reason for this change?

Reply:
It's just that during my testing, the Discord client didn't respond and I was wondering if adding commands might have failed. I found out that restarting the client would fix the issue
If you mind, I'll change it back

Reply:
I will do that and update code style as well.